src/UkModulusCheck/Implementation.fs

```diff
@@ -0,0 +1,232 @@
+                    // Perform the double alternate check except: Add 27 to the total (ie before you divide by 10)
+                    (doubleAlternateModulus rule.Weightings number + 27) % 10 = 0
+
+                | Exception 4 ->
+                    // Perform the standard modulus 11 check.
+                    // After you have finished the check, ensure that the remainder is the same as the two-digit checkdigit;
+                    // the checkdigit for exception 4 is gh from the original account number.
+                    let modulus = standardModulus rule.Weightings number % 11
+                    let checkdigit = (char2int number.[Position.G]) * 10 + (char2int number.[Position.H])
+                    modulus = checkdigit
+
+                | Exception 7 ->
+                    // Perform the check as specified, except if g = 9 zeroise weighting positions u-b.
+                    let rule', number' =
+                        if number.[Position.G] = '9'
+                        then { rule with Weightings = rule.Weightings.[Position.B+1..] }, number.[Position.B+1..]
+                        else rule, number
+                    validateRule rule' number'
+
+                | Exception 8 ->
+                    // Perform the check as specified, except substitute the sorting code with 090126, for check purposes only.
+                    validateRule rule ("090126" + number.[Position.A..])
+
+                | Exception 14 ->
+                    // Perform the modulus 11 check as normal: If the check passes (that is, there is no remainder), then the account number should be considered valid. Do not perform the second check.
+                    // If the first check fails, then the second check must be performed as specified below.
+                    let secondCheck() =
+                        // If the 8th digit of the account number (reading from left to right) is not 0, 1 or 9 then the account number fails the second check and is not a valid Coutts account number.
+                        // If the 8th digit is 0, 1 or 9, then remove the digit from the account number and insert a 0 as the 1st digit for check purposes only.
+                        // Perform the modulus 11 check on the modified account number using the same weightings as specified in the table.
+                        // - If there is no remainder, then the account number should be considered valid.
+                        // - If there is a remainder, then the account number fails the second check and is not a valid Coutts account number
+                        if "019" |> Seq.contains accountNo.[7]
+                        then validateRule rule (sortCode + "0" + accountNo.[..6])
+                        else false
+
+                    validateRule rule number || secondCheck()
+
+                | _ -> false // TODO: Should we fail for unrecognized exception?
+
+            | [rule1; rule2], [None; Some ex2] ->
+                match ex2 with
```

Comment:
This internal pattern matching can be flatted to | [rule], [Some (Exception 1)] etc, I think I having it on the top level seems more explicit.